contracts/CommunityOwnable.sol

```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: Mozilla Public License 2.0
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+contract CommunityOwnable {
+    error CommunityOwnable_InvalidTokenAddress();
+    error CommunityOwnable_NotAuthorized();
+
+    address public immutable ownerToken;
+    address public immutable masterToken;
+
+    constructor(address _ownerToken, address _masterToken) {
+        ownerToken = _ownerToken;
+        masterToken = _masterToken;
+
+        if (ownerToken == address(0) && masterToken == address(0)) {
+            revert CommunityOwnable_InvalidTokenAddress();
+        }
+    }
+
+    modifier onlyCommunityOwner() {
+        if (
+            (ownerToken != address(0) && IERC721(ownerToken).balanceOf(msg.sender) == 0)
+                && (masterToken != address(0) && IERC721(masterToken).balanceOf(msg.sender) == 0)
+        ) {
+            revert CommunityOwnable_NotAuthorized();
+        }
+        _;
+    }
+}
```

Comment:
This might be a good time to add missing natspec comments.
If you need inspiration, checkout `CommunityTokenDeployer.sol`